capstone/src/main/resources/static/user.js

```diff
@@ -430,13 +440,27 @@ function createCalendarEventElement(event, eventTime) {
 
 /* function to create a public profile of an organization */
 function getPublicProfile() {
+ showSpinner();
  var organizationId = window.location.hash.substring(1);
  fetch('get-public-profile?organization-id=' + organizationId).then(response => response.json()).then((data) => {
    createProfile(data, false, true);
    const eventDiv = document.getElementById("hosted-events");
    data.events.forEach((event) => eventDiv.appendChild(createSavedEventElement(event, true, false)));
    document.getElementById("public-image-a").setAttribute("href", "get-public-image?email=" + data.email);
    document.getElementById("public-image-img").setAttribute("src", "get-public-image?email=" + data.email);
+  hideSpinner();
  });
 }
 
+/* displays spinner on the page and hide main content */
+function showSpinner() {
+  document.getElementById('contents').style.display="none";
```

Comment:
Instead of hiding all of the main page content while the loading is in progress, can you just make the spinner float on top of the existing content using something like CSS `position: absolute`?
I think this would reduce the feeling of "flicker" when the page is blank while loading is happening.